src/lib/Libifl/pbs_loadconf.c

```diff
@@ -582,18 +583,13 @@ __pbs_loadconf(int reload)
 				pbs_conf.pbs_conf_remote_viewer = strdup(conf_value);
 			}
 #endif
-#ifndef WIN32
 			else if (!strcmp(conf_name, PBS_CONF_AUTH)) {
-				if (!strcasecmp(conf_value, "MUNGE")) {
-				   pbs_conf.auth_method = AUTH_MUNGE;
-				} else if (!strcasecmp(conf_value, "GSS")) {
-					pbs_conf.auth_method = AUTH_GSS;
-				} else {
-					fprintf(stderr, "pbsconf error: illegal value for %s\n",PBS_CONF_AUTH);
-					goto err;
+				int i = 0;
+				memset(pbs_conf.auth_method, '\0', sizeof(pbs_conf.auth_method));
+				for (i = 0; i < strlen(conf_value); i++) {
+					pbs_conf.auth_method[i] = tolower(conf_value[i]);
```

Comment:
How will we validate "pbs_conf.auth_method" is valid supported auth method (like GSS, Munge or any future authentication method) ?

Reply:
if the value is not resvport then it will go into load_auth_lib() and there it will try to load .so file, if the file doesn't exist or not an libauth library then it will fail, which eventually considered as not supported auth method. Means if load_auth_lib() succeeds then its supported method.